clang/lib/Sema/SemaFunctionEffects.cpp

```diff
@@ -78,22 +78,19 @@ class ViolationSite {
 // be inferred as holding that effect.
 struct Violation {
   FunctionEffect Effect;
-  FunctionEffect
+  std::optional<FunctionEffect>
       CalleeEffectPreventingInference; // Only for certain IDs; can be None.
```

Comment:
```suggestion
      CalleeEffectPreventingInference; // Only for certain IDs; can be nullopt.
```
Comment still needs updating